File: src/com/stephenclhc/SearchInsertPosition.java
package com.stephenclhc;

public class SearchInsertPosition {
    // 35. Search Insert Position

    public int searchInsert(int[] nums, int target){
        int leftIndex = 0;
        int rightIndex = nums.length;
        int middleIndex;

        if(target > nums[nums.length - 1])
            return nums.length;

        while (leftIndex <= rightIndex){
            middleIndex = leftIndex + (rightIndex - leftIndex) / 2;
            if (nums[middleIndex] == target)
                return middleIndex;
            if (nums[middleIndex] < target)
                leftIndex = middleIndex + 1;
            else rightIndex = middleIndex - 1;
        }
        return leftIndex;
    }

}

File: JavaScript/search-insert-position.js
// 35. Search Insert Position

var searchInsert = function(nums, target) {
    let left = 0;
    let right = nums.length - 1;

    while (left <= right){
        let mid = left + Math.floor((right - left) / 2);
        if (target > nums[mid - 1] && target <= nums[mid]) return mid;
        if(target < nums[left]) return left;
        if(target > nums[right]) return right + 1;
        if (target > nums[mid]) left = mid + 1;
        else right = mid - 1;
    }
    return -1;
};